### PR TITLE
Pulled service inline with python version

### PR DESCRIPTION
**Pull request by morgangraphics**
Added 
nulls & prune options
Split major functionality into classes
Updated and added tests
Utf-8 text conversion on names e.g. \u00c4kr\u00e4s to Äkräs
Locale aware sorting
Empty values are converted to nulls to be aligned with JSON spec

### CODE REVIEW

File: components/files.js [hunk 1]
```diff
@@ -6,127 +6,297 @@ const firstBy = require('thenby');
...
+
+  /**
+   * Custom sorting function that allows for sorting None while also maintaining
+   * locale aware sorting config["nulls"] will allow for sorting None and putting
+   * them at the front or end of the list
+   * l = [1, 3, 2, 5, 4, null, 7]
+   * console.log('Last = ', sort_results(l)) e.g nulls=last
+   * Last = [1, 2, 3, 4, 5, 7, None]
+   * console.log('First = ', sort_results(l)) e.g nulls=first
+   * First = [None, 1, 2, 3, 4, 5, 7]
```
Comment (morgangraphics) [inline components/files.js:239]: None should be null

File: components/files.js [hunk 2]
```diff
@@ -6,127 +6,297 @@ const firstBy = require('thenby');
...
+    return this.getData();
+  }
+
+  /**
+   * Custom sorting function that allows for sorting None while also maintaining
+   * locale aware sorting config["nulls"] will allow for sorting None and putting
+   * them at the front or end of the list
+   * l = [1, 3, 2, 5, 4, null, 7]
+   * console.log('Last = ', sort_results(l)) e.g nulls=last
+   * Last = [1, 2, 3, 4, 5, 7, None]
```
Comment (morgangraphics) [inline components/files.js:237]: None should be null

File: components/files.js [hunk 3]
```diff
@@ -6,127 +6,297 @@ const firstBy = require('thenby');
...
+    // If there are more than one item then we dynamically sort on those
+    for (let i = 1; i < this.config.s.length; i += 1) {
+      srt = srt.thenBy(this.config.s[i].column,
+        { cmp: this.sortIl8nStr.bind(this), direction: this.config.s[i].sort });
+    }
+    return results.sort(srt);
+  }
+
+  /**
+   * Sorting compare function to handle
```
Comment (morgangraphics) [inline components/files.js:257]: The functionality here is explained in the method above. Does it make sense to detail it here instead?

**Comment by heronblade** [comment]
@morgangraphics , how do you get past the ssl issue with Postman?  I can't run the tests.  I can `curl` if I add `-k`, I can view every thing via the browser.  It isn't a blocker, just a setup question

File: api/dc.js [hunk 4]
```diff
@@ -51,18 +85,32 @@ const dcPostBase = {
...
-  path: '/dc/{character}',
+  path: '/dc/{characters}',
   options: {
     handler: (req, handlr) => {
-      const options = Object.assign({}, req.query, req.payload, req.params, { universe: 'dc' });
-      const config = api.handleConfig(options);
-      return api.handleRequest(config, handlr);
+      const options = Object.assign({}, req.query, req.payload, req.params, { universe });
+      let response;
+      if ('help' in options) {
```
Comment (heronblade) [inline api/dc.js:93]: this is super nice. love being able to get help this way

File: components/files.js (hunk 3 above)
Comment (heronblade) [reply components/files.js:258]: i think it's fine in the above method

**Comment by heronblade** [comment]
Now, this might be a dumb question, but why have a `api/dc.js` and an `api.marvel.js` instead of a `superhero.js` or maybe `entry.js` which will take path of `/dc` or `/marvel` depending on what you're querying?  Then lines 6-8 on both `api/dc.js` and `api/marvel.js` will rely on whatever is being queried.  Wouldn't that make it easier for you to spread this out to consume more than just the two superhero files?

Not sure if this is even a good idea or something that's worth bringing up. Otherwise, everything else looks great. I was just looking for something to add here :)

**Comment by morgangraphics** [comment]
Your question about a single entry way is 100% perfectly valid and something that was brought up in the Python version. I wanted to be very explicit in defining the two entry points. So one compromise was to make the Classes function the way they do with the universe abstractions. Lastly, there is a 3rd file which contains both universes and there could potentially be more universes from Image, Darkhorse etc. As the service scales, being explicit becomes less maintainable. However, with just 2 endpoints, I think the compromise is reasonable. 

To run the tests in Postman with SSL  - go to Settings and turn off SSL certificate verification tab.

File: api/dc.js (hunk 4 above)
Comment (morgangraphics) [reply api/dc.js:93]: Thanks man. It's very Pythonic actually.

**Pull request merged**